templates/res/js/dashboard.js

```diff
@@ -21,14 +34,68 @@ CRM.$(function($) {
     updateCharsLeft();
   });
 
+  cj('#make-post').submit(function(e) {
+    e.preventDefault();
+
+    // Begin validation
+    if (cj('#post-content').val().length === 0) {
+      showError('Post cannot be empty.');
+      return;
+    }
+
+    if (cj('#chars-left').html() < 0) {
+      showError('Please limit the post length to ' + post_char_limit + ' characters.');
+      return;
+    }
+
+    if (!cj('#post-to-facebook').is(':checked') && !cj('#post-to-twitter').is(':checked')) {
+      showError('Please check at least one social network.');
+      return;
+    }
+
+    if (xhrOn) {
+      return;
+    }
+
+    var postData = {};
+    cj('#make-post').find('[name]').each(function() {
+      if (cj(this).is(':checkbox')) {
+        if (cj(this).is(':checked')) {
+          postData[this.name] = cj(this).is(':checked'); 
+        }
+      }
+      else {
+        postData[this.name] = this.value;  
+      }
+    });
+
+    var xhrOn = true;
+    cj('#post-button').val('Posting..');
+
+    CRM.api3('CivisocialUser', 'updatestatus', postData).done(function(result) {
+      console.log(result);
+      if (result.is_error) {
+        showError(result.error_message);
+      }
+      else {
+        cj('#post-button').val('Posted!');
+        cj('#post-content').val('');
+        updateCharsLeft();
+      }
+      xhrOn = false;
+    });
+  });
+
   function updateCharsLeft() {
     var post_char_left = post_char_limit - cj('#post-content').val().length;
     cj('#chars-left').html(post_char_left);
   }
 
-  cj('#make-post').submit(function(e) {
-    e.preventDefault();
-
-    // Pefrom AJAX request to API
-  });
+  function showError(message) {
+    cj('#post-content').addClass('error');
+    if (cj('#make-post > span.error')) {
+      cj('#make-post > span.error').remove();
+    }
+    cj('<span class="error block">' + message + '</span>').insertAfter('#make-post > .post-to');
+  }
```

Comment:
This is an outdated pattern. We no longer use cj and we try to avoid putting functions or variables in the global scope.
The way to update this code would be to wrap the whole thing in CRM.$(function($) { ... }); and replace all cj with $.